R/wlr.R

```diff
@@ -85,8 +86,34 @@
 #'
 #' # Example 3: WLR test with early zero wights
 #' x |> wlr(weight = early_zero(early_period = 4))
+#'
+#' # For increased computational speed when running many WLR tests, you can
+#' # pre-compute the counting_process() step first, and then pass the result of
+#' # counting_process() directly to wlr()
+#' x <- x |> counting_process(arm = "experimental")
+#' x |> wlr(weight = fh(rho = 0, gamma = 1))
+#' x |> wlr(weight = mb(delay = 4, w_max = 2))
+#' x |> wlr(weight = early_zero(early_period = 4))
 wlr <- function(data, weight, return_variance = FALSE) {
+  UseMethod("wlr", data)
+}
+
+#' @rdname wlr
+#' @export
+wlr.counting_process <- function(data, weight, return_variance = FALSE) {
+  wlr.default(data, weight, return_variance = FALSE)
+}
+
+#' @rdname wlr
+#' @export
+wlr.tte_data <- function(data, weight, return_variance = FALSE) {
   x <- data |> counting_process(arm = "experimental")
+  wlr.default(x, weight, return_variance = FALSE)
+}
+
+#' @export
```

Comment:
Thank you, @jdblischak! I like the idea of `wlr.default` to prevent code duplication. I have a minor comment - should we consider not exporting wlr.default since it is rarely used by users?

Reply:
`wlr.default` isn't actually exported in `NAMESPACE`. Using `@export` for an S3 method is a roxygen2 convention. It only results in `S3method(wlr,default)` being added to `NAMESPACE`. In other words, the user won't directly see it in the R console, as I demonstrate below:

```R
simtrial::wlr.default
## Error: 'wlr.default' is not an exported object from 'namespace:simtrial'
```

If I remove `@export` from `wlr.default()`, then roxygen2 complains because this would break the S3 dispatch:

```
devtools::document(roclets = c('rd', 'collate', 'namespace'))
## ℹ Updating simtrial documentation
## ℹ Loading simtrial
## ✖ wlr.R:114: S3 method `wlr.default` needs @export or @exportS3method tag.
## Writing 'NAMESPACE'
## Documentation completed
```

Reply:
However, if `wlr.default()` is distracting to developers, I can remove it. This would require users to always pass `"tte_data"` or `"counting_process"`, which I believe is what we want.

Reply:
Good to know. I am okay with keeping `wlr.default`!

Reply:
I removed `wlr.default()` in https://github.com/Merck/simtrial/pull/276/commits/de83a4515703a7b9cb23e4aad90fd58d3d9088b0. If we decide we want it a default method in the future (eg so that we can pass non-classed objects to `wlr()`), we can always add it then